client/src/main/java/com/alibaba/nacos/client/naming/core/HostReactor.java

```diff
@@ -161,7 +163,8 @@ public void unSubscribe(String serviceName, String clusters, EventListener event
     public ServiceInfo processServiceJson(String json) {
         ServiceInfo serviceInfo = JacksonUtils.toObj(json, ServiceInfo.class);
         ServiceInfo oldService = serviceInfoMap.get(serviceInfo.getKey());
-        if (serviceInfo.getHosts() == null || !serviceInfo.validate()) {
+        
+        if (pushEmptyProtection && !serviceInfo.validate()) {
```

Comment:
should we judge null for serviceInfo.getHosts?
It come from json, and might is null, if hosts is null. the validate method may throw NPE.

Reply:
the hosts cann't be null, it's instance variable.

Reply:
hosts is not final. And the set method can directly set it.
If the json return "{hosts: null}", it will be null

Reply:
right, should be check. Can't guarantee the server side response value.